source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,707 @@
+
+- ``OpenUploadStreamAsync()`` or ``OpenUploadStream()``: Opens a new upload stream to which
+  you can write file contents
+- ``UploadFromStreamAsync()`` or ``UploadFromStream()``: Uploads the contents of an existing
+  stream to a GridFS file
+
+The following sections describe these methods in more detail.
+
+.. _gridfs-open-upload-stream:
+
+Write to an Upload Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``OpenUploadStreamAsync()`` or ``OpenUploadStream()`` method to create an upload
+stream for a given file name. These methods accept the following parameters:
+
+.. list-table::
+
+   * - Parameter
+     - Description
+   * - ``filename``
+     - | The name of the file to upload.
+       | 
+       | **Data type**: {+string-data-type+}
+   * - ``options``
+     - | *Optional.* An instance of the ``GridFSUploadOptions`` class that specifies the
+         configuration for the upload stream. The default value is ``null``.
+       | 
+       | **Data type**: `GridFSUploadOptions <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.GridFS.GridFSBucket.GridFSUploadOptions.html>`__
+   * - ``cancellationToken``
+     - | *Optional.* A token that you can use to cancel the operation.
+       | 
+       | **Data type**: `CancellationToken <https://learn.microsoft.com/dotnet/api/system.threading.cancellationtoken>`__
+
+The following code example demonstrates how to open an upload stream by performing the
+following steps:
+
+- Calls the ``OpenUploadStream()`` method to open a writable GridFS stream for a new file
+  named ``"my_file"``
+- Calls the ``Write()`` method to write data to ``my_file``
+- Calls the ``Close()`` method to close the stream that points to ``my_file`` 
+
+Select the :guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: open-upload-stream-async
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-open-upload-stream-async
+         :end-before: // end-open-upload-stream-async
+
+   .. tab:: Synchronous
+      :tabid: open-upload-stream-sync
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-open-upload-stream
+         :end-before: // end-open-upload-stream
+
+To customize the upload stream configuration, pass an instance of the
+``GridFSUploadOptions`` class to the ``OpenUploadStreamAsync()`` or ``OpenUploadStream()``
+method. The ``GridFSUploadOptions`` class contains the following properties:
+
+.. list-table::
+
+   * - Property
+     - Description
+   * - ``BatchSize``
+     - | The number of chunks to upload in each batch. The default value is ``16777216``
+         divided by the value of the following property, ``ChunkSizeBytes``. 
+       | 
+       | **Data type**: ``int?`` 
+       
+   * - ``ChunkSizeBytes``
+     - | The maximum size of each chunk.
```

Comment:
S: add a default

Reply:
Can't find this in the repo. Will leave a note for the tech reviewer.